test/quantization/pt2e/test_quantize_pt2e.py

```diff
@@ -903,6 +903,124 @@ def validate(self, model: torch.fx.GraphModule) -> None:
             m, expected_node_list=node_list, expected_node_occurrence=node_occurrence
         )
 
+    def test_derived_qspec_per_channel(self):
+        class BackendAQuantizer(Quantizer):
+            def annotate(self, model: torch.fx.GraphModule) -> torch.fx.GraphModule:
+                for node in model.graph.nodes:
+                    if (
+                        node.op == "call_function"
+                        and node.target == torch.ops.aten.convolution.default
+                    ):
+                        input_act = node.args[0]
+                        assert isinstance(input_act, Node)
+                        weight = node.args[1]
+                        assert isinstance(weight, Node)
+                        bias = node.args[2]
+                        assert isinstance(bias, Node)
+                        act_qspec = QuantizationSpec(
+                            dtype=torch.uint8,
+                            quant_min=0,
+                            quant_max=255,
+                            qscheme=torch.per_tensor_affine,
+                            is_dynamic=False,
+                            observer_or_fake_quant_ctr=observer.default_observer,
+                        )
+                        weight_qspec = QuantizationSpec(
+                            dtype=torch.int8,
+                            quant_min=-128,
+                            quant_max=127,
+                            qscheme=torch.per_tensor_affine,
```

Comment:
what you really want to test is have weight qspec with per_channel_affine and then have derived_qparams_fn, return per channel scale tensor. That way the bias zp and scale tensors are of size = output channels

Reply:
ah sorry, I'll update it in the capture diff